packages/mui-docs/src/MarkdownElement/MarkdownElement.tsx

```diff
@@ -803,18 +804,19 @@ const Root = styled('div')(
   }),
 );
 
-function handleClick(event: Event) {
+function handleHeaderClick(event: Event) {
   const selection = document.getSelection();
 
   if (selection === null) {
     return;
   }
-  const isRangeSelection = selection.type === 'Range';
 
-  if (isRangeSelection) {
+  if (selection.type === 'Range') {
+    // Disable the <a> behavior to be able to select text.
```

Comment:
We have this implicit rule to try to document each `event.preventDefault();` call in the source because they tend to have strong implications, it's hard to not break stuff with them.